test/modules/http/intercept/client-request.test.ts

```diff
@@ -0,0 +1,67 @@
+import { vi, it, expect, beforeAll, afterEach, afterAll } from 'vitest'
+import http from 'http'
+import { HttpServer } from '@open-draft/test-server/http'
+import { ClientRequestInterceptor } from '../../../../src/interceptors/ClientRequest'
+import { REQUEST_ID_REGEXP, waitForClientRequest } from '../../../helpers'
+import { RequestController } from '../../../../src/RequestController'
+import { HttpRequestEventMap } from '../../../../src/glossary'
+
+const httpServer = new HttpServer((app) => {
+  app.get('/user', (req, res) => {
+    res.status(200).send('user-body')
+  })
+})
+
+const resolver = vi.fn<HttpRequestEventMap['request']>()
+
+const interceptor = new ClientRequestInterceptor()
+interceptor.on('request', resolver)
```

Comment:
I actually grew to dislike this pattern. Sorry it's still present in the test suite. It leads to a shared state in a shape of the `resolver` function, and that's really bad. 

May I please ask you to add the request listener within the test instead? It will yield a much simpler setup. Thanks!

Reply:
Sure! like this?
![image](https://github.com/user-attachments/assets/045a04a5-474b-49e8-b9ac-d71a69bb72f6)

Reply:
I think so! 

And please add `interceptor.removeAllListeners()` to the `afterEach()` hook.